src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/OnTypeFormattingHandler.cs

```diff
@@ -71,7 +71,9 @@ public async Task<TextEdit[]> HandleRequestAsync(DocumentOnTypeFormattingParams
                 return null;
             }
 
-            var projectionResult = await _projectionProvider.GetProjectionAsync(documentSnapshot, request.Position, cancellationToken).ConfigureAwait(false);
+            // For onTypeFormatting, it makes more sense to look up the projection of the character that was inserted. Aka. request.Position.Character - 1
+            var position = new Position(request.Position.Line, request.Position.Character - 1);
```

Comment:
Don't think so. `request.Position.Character` is set to the position after the trigger character. Since the trigger character length can't be zero, `request.Position.Character` cannot be 0.